app/code/Magento/GoogleGtag/Block/Ga.php

```diff
@@ -0,0 +1,198 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\GoogleGtag\Block;
+
+use Magento\Cookie\Helper\Cookie;
+use Magento\Framework\View\Element\Template;
+use Magento\Framework\View\Element\Template\Context;
+use Magento\GoogleGtag\Helper\GtagConfiguration;
+use Magento\Sales\Model\ResourceModel\Order\CollectionFactory;
+use Magento\Store\Model\ScopeInterface;
+
+/**
+ * GoogleAnalytics Page Block
+ *
+ * @api
+ */
+class Ga extends Template
+{
+    /**
+     * @var GtagConfiguration
+     */
+    private $googleGtagConfig;
+
+    /**
+     * @var CollectionFactory
+     */
+    private $salesOrderCollection;
+
+    /**
+     * @var Cookie
+     */
+    private $cookieHelper;
+
+    /**
+     * @param Context $context
+     * @param CollectionFactory $salesOrderCollection
+     * @param GtagConfiguration $googleGtagConfig
+     * @param array $data
+     * @param Cookie $cookieHelper
+     */
+    public function __construct(
+        Context $context,
+        CollectionFactory $salesOrderCollection,
+        GtagConfiguration $googleGtagConfig,
+        Cookie $cookieHelper,
+        array $data = []
+    ) {
+        $this->googleGtagConfig = $googleGtagConfig;
+        $this->salesOrderCollection = $salesOrderCollection;
+        $this->cookieHelper = $cookieHelper;
+        parent::__construct($context, $data);
+    }
+
+    /**
+     * Get config
+     *
+     * @param string $path
+     * @return mixed
+     */
+    public function getConfig($path): string
```

Comment:
This method seems to have too big of a responsibility to fetch any configuration. I would suggest wrapping it to use more granular approach and fetch only required config entries.